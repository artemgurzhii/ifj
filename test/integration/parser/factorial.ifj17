function factorial(n as integer) as integer
  dim temp_result as integer
  dim decremented_n as integer
  dim result as integer

  if n < 2 then
    result = 1
  else
    decremented_n = n - 1
    temp_result = factorial(decremented_n)
    result = n * temp_result
  end if

  return result
end function
