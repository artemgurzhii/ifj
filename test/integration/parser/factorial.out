(function factorial -> (id integer)
  (decl
    (id n)
    : (id integer))
  (dim
    (decl
      (id temp_result)
      : (id integer)))
  (dim
    (decl
      (id decremented_n)
      : (id integer)))
  (dim
    (decl
      (id result)
      : (id integer)))
  (if (< (id n) (int 2))
    (= (id result) (int 1)))
  (else
    (= (id decremented_n) (- (id n) (int 1)))
    (= (id temp_result) (call
      (id factorial)
      (id decremented_n)))
    (= (id result) (* (id n) (id temp_result))))
  (return
    (id result)))
